pkg/skaffold/jib/jib.go

```diff
@@ -17,32 +17,114 @@ limitations under the License.
 package jib
 
 import (
+	"encoding/json"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"sort"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/karrick/godirwalk"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func getDependencies(cmd *exec.Cmd) ([]string, error) {
-	stdout, err := util.RunCmdOut(cmd)
-	if err != nil {
-		return nil, err
+// filesTemplate contains fields parsed from Jib's JSON output
+type filesTemplate struct {
+	Build  []string
+	Inputs []string
+	Ignore []string
+}
+
+// filesLists contains cached build/input dependencies
+type filesLists struct {
+	WatchedBuildFiles []string
+	WatchedInputFiles []string
+	BuildFileTimes    map[string]time.Time
+}
+
+// watchedFiles maps from project name to watched files
+var watchedFiles = map[string]filesLists{}
+
+func getDependencies(cmd *exec.Cmd, projectName string) ([]string, error) {
+	if len(watchedFiles[projectName].WatchedInputFiles) == 0 && len(watchedFiles[projectName].WatchedBuildFiles) == 0 {
+		// Make sure build file modification time map is setup
+		if watchedFiles[projectName].BuildFileTimes == nil {
+			watched := watchedFiles[projectName]
+			watched.BuildFileTimes = make(map[string]time.Time)
+			watchedFiles[projectName] = watched
+		}
+
+		// Refresh dependency list if empty
+		if err := refreshDependencyList(cmd, projectName); err != nil {
+			return nil, err
+		}
+	} else {
+		// Refresh dependency list if any build definitions have changed
+		for _, buildFile := range watchedFiles[projectName].WatchedBuildFiles {
```

Comment:
What happens if I add a new input file?  Or a new build file?  (For example, if we have a `.mvn` directory at the root, jib should return `.mvn` as a build file.)